x-pack/plugins/apm/public/components/shared/charts/transaction_distribution_chart/index.tsx

```diff
@@ -75,22 +76,21 @@ const CHART_PLACEHOLDER_VALUE = 0.0001;
 // as a circular marker instead of a bar
 // This provides a workaround by making the next bin not empty
 export const replaceHistogramDotsWithBars = (
-  originalHistogram: HistogramItem[] | undefined
+  originalHistogram: HistogramItem[]
 ) => {
-  if (originalHistogram === undefined) return;
   const histogram = [...originalHistogram];
-  {
-    for (let i = 0; i < histogram.length - 1; i++) {
-      if (
-        histogram[i].doc_count > 0 &&
-        histogram[i].doc_count !== CHART_PLACEHOLDER_VALUE &&
-        histogram[i + 1].doc_count === 0
-      ) {
-        histogram[i + 1].doc_count = CHART_PLACEHOLDER_VALUE;
-      }
+
+  for (let i = 0; i < histogram.length - 1; i++) {
```

Comment:
Can we rewrite this from a `for`. Preferably replace it with `map`, `filter` or `reduce`

Reply:
Refactored this to use `reduce` in 0a56509.